import/webview/plugin.cpp

```diff
@@ -67,13 +67,40 @@ void SailfishOSWebViewPlugin::initializeEngine(QQmlEngine *engine, const char *u
 
 RawWebView::RawWebView(QQuickItem *parent)
     : QuickMozView(parent)
+    , m_vkbMargin(0.0)
 {
 }
 
 RawWebView::~RawWebView()
 {
 }
 
+qreal RawWebView::virtualKeyboardMargin() const
+{
+    return m_vkbMargin;
+}
+
+void RawWebView::setVirtualKeyboardMargin(qreal vkbMargin)
+{
+    if (m_vkbMargin != vkbMargin) {
+        qreal delta = vkbMargin - m_vkbMargin;
+        m_vkbMargin = vkbMargin;
+        QMargins m = margins();
+        m.setBottom(m.bottom() + delta);
+        setMargins(m);
+        emit virtualKeyboardMarginChanged();
+
+        QVariantMap map;
+        map.insert("imOpen", m_vkbMargin > 0);
+        map.insert("pixelRatio", SailfishOS::WebEngineSettings::instance()->pixelRatio());
+        map.insert("bottomMargin", m.bottom());
+        map.insert("screenWidth", width());
+        map.insert("screenHeight", height());
```

Comment:
To clarify: when opening a normal text input method VKB, height() here either returns 700 (with vkbMargin 580) or 700.56 (with vkbMargin 0) rather than 700 or 1280 (which is strange, but I guess it's an order-of-evaluation issue with QML).  But, when opening a numeric input method VKB, it can return different values such that height()+vkbMargin can be 1280, 1300, 1488, etc different values.  I don't understand why, but I guess it may not matter.